README.md

```diff
@@ -1 +1,186 @@
+[david-dm-dev-image]:https://img.shields.io/david/dev/moxystudio/react-wait-for-react.svg
+
+Easily render a loader while your React app is loading, optionally waiting for a promise as well.
+
+## Installation
+
+```sh
+$ npm install @moxy/react-wait-for-react
+```
+
+This library is written in modern JavaScript and is published in both CommonJS and ES module transpiled variants. If you target older browsers please make sure to transpile accordingly.
+
+## Motivation
+
+Certain apps or pages have impactful experiences. These experiences can make the total bundle size larger as they pack possibly large dependencies and media assets, such as 3D objects and audio files.
+
+It's then often normal to preload all the required files for an uninterrupted experience. `@moxy/wait-for-react` is a library that makes it easy to display a loader before your static or server-side rendered app becomes interactive, and optionally until all the required files are loaded (via a promise).
+
+<img src="https://developers.google.com/web/fundamentals/performance/images/perf-metrics-load-timeline.png" alt="Performance metrics timeline" width="700" />
+
+⚠️ You should still render the app or page contents "below" the loader, to keep your website SEO friendly.
+
+## Demo
+
+You may see a simple demo of `react-wait-for-react` in [https://moxystudio.github.io/react-wait-for-react](https://moxystudio.github.io/react-wait-for-react/).
+
+## Usage
+
+Using `<WaitForReact>` to render a progress bar while your page assets are being loaded:
+
+```js
+import React from 'react';
+import classNames from 'classnames';
+import WaitForReact from '@moxy/react-wait-for-react';
+import styles from './MyPage.module.css';
+
+const preloadAssets = async () => {
+    // Preload files, like a mp3, 3d objects, etc..
+};
+
+const MyPage = () => {
+    const promise = useMemo(() => preloadAssets(), []);
+    const applyProgressBeforeInteractive = useCallback((elements, progress) =>
+        elements.progressBar.style.transform = `scaleX(${progress})`);
+
+    return (
+        <main>
+            <WaitForReact
+                applyProgressBeforeInteractive={ applyProgressBeforeInteractive }
+                promise={ promise }>
+                { ({ progress }) => (
+                    <div
+                        data-wait-for-react-element="progressBar"
+                        className={ classNames(styles.progressBar, progress > 1 && styles.done) }
+                        style={ { transform: `scaleX(${progress})` } } />
+                ) }
+            </WaitForReact>
+            <div>My Awesome Page</div>
+        </main>
+    );
+};
+
+export default MyPage;
+```
+
+## API
+
+This package exports a single component called `<WaitForReact>`, with the following props:
+
+### maxProgressBeforeInteractive
+
+Type: `number`   
+Default: `0.4`
+
+The maximum value the progress can take before the app becomes interactive. Takes a value between 0 and 1 (exclusive).
```

Comment:
I don't think I quite understand the description here.

We are trying to say that even if the app becomes interactive immediately, the progress bar still grows to 0.4?

Reply:
No, what this means is that if you are in a slow network and it's taking long to download the necessary JS chunks for your app to become interactive, the progress bar will ONLY grow until `maxProgressBeforeInteractive` (0.4) by default.

If your app becomes interactive almost instantly, the progress bar will grow almost instantly too.

if you have any suggestion in terms of english to make it clear, let me know